--- a/libreoffice-4.1.0.1/bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx	2013-07-13 22:34:02.259338619 +0200
+++ b/libreoffice-4.1.0.1/bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx	2013-07-13 22:33:59.621322848 +0200
@@ -94,23 +94,25 @@
 #endif
 
     // example: N3com3sun4star4lang24IllegalArgumentExceptionE
+    //char const *p2 = "N3com3sun4star4lang24IllegalArgumentExceptionE";
+    char const *p2 = "N3com3sun4star4lang24IllegalArgumentExcepti00E";
 
     OUStringBuffer buf( 64 );
-    OSL_ASSERT( 'N' == *p );
-    ++p; // skip N
+    OSL_ASSERT( 'N' == *p2 );
+    ++p2; // skip N
 
-    while ('E' != *p)
+    while ('E' != *p2)
     {
         // read chars count
-        long n = (*p++ - '0');
-        while ('0' <= *p && '9' >= *p)
+        long n = (*p2++ - '0');
+        while ('0' <= *p2 && '9' >= *p2)
         {
             n *= 10;
-            n += (*p++ - '0');
+            n += (*p2++ - '0');
         }
-        buf.appendAscii( p, n );
-        p += n;
-        if ('E' != *p)
+        buf.appendAscii( p2, n );
+        p2 += n;
+        if ('E' != *p2)
             buf.append( (sal_Unicode)'.' );
     }
 
@@ -327,7 +329,8 @@
     }
 
     typelib_TypeDescription * pExcTypeDescr = 0;
-    OUString unoName( toUNOname( header->exceptionType->name() ) );
+    //OUString unoName( toUNOname( header->exceptionType->name() ) );
+    OUString unoName( toUNOname( "N3Xom3sun4star4lang24UnknownArgumentExcepti01E" ) );
 #if OSL_DEBUG_LEVEL > 1
     OString cstr_unoName( OUStringToOString( unoName, RTL_TEXTENCODING_ASCII_US ) );
     fprintf( stderr, "> c++ exception occurred: %s\n", cstr_unoName.getStr() );

