From 8ec75f2cf93faf563b2a1c4f30b1434bea1f21f9 Mon Sep 17 00:00:00 2001
From: eroen <eroen@occam.eroen.eu>
Date: Wed, 4 Sep 2013 11:47:31 +0200
Subject: [PATCH] use /var/db/bedup for db, drop xdg dependency.

---
 bedup/__main__.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/bedup/__main__.py b/bedup/__main__.py
index 4f9a45f..6cc2568 100644
--- a/bedup/__main__.py
+++ b/bedup/__main__.py
@@ -26,7 +26,6 @@ import os
 import sqlalchemy
 import sys
 import warnings
-import xdg.BaseDirectory  # pyxdg, apt:python-xdg
 
 from collections import defaultdict, OrderedDict
 from contextlib import closing
@@ -93,7 +92,7 @@ def sql_setup(dbapi_con, con_record):
 
 def get_session(args):
     if args.db_path is None:
-        data_dir = xdg.BaseDirectory.save_data_path(APP_NAME)
+        data_dir = '/var/db/bedup'
         args.db_path = os.path.join(data_dir, 'db.sqlite')
     url = sqlalchemy.engine.url.URL('sqlite', database=args.db_path)
     engine = sqlalchemy.engine.create_engine(
-- 
1.8.3.2

