From 1a928b40e7f071fb31b182809db15ac037a0f48a Mon Sep 17 00:00:00 2001
From: eroen <eroen@occam.eroen.eu>
Date: Mon, 22 Oct 2012 09:33:55 +0200
Subject: [PATCH 5/5] don't build static tinyxml as lib

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86c2c76..54d4570 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,7 @@ SET(PROJECT_SRCS
     VegetationConfiguration.cpp
     main.cpp
 )
+FILE(GLOB TINYXML_SOURCES "${dfhack-tinyxml_SOURCE_DIR}/*.cpp")
 
 IF(UNIX)
 	IF(APPLE)
@@ -158,7 +159,7 @@ ELSE(UNIX)
         ${stonesense_SOURCE_DIR}/allegrow/lib
     )
 ENDIF(UNIX)
-DFHACK_PLUGIN(stonesense ${PROJECT_SRCS} LINK_LIBRARIES dfhack-tinyxml ${PROJECT_LIBS})
+DFHACK_PLUGIN(stonesense ${PROJECT_SRCS} ${TINYXML_SOURCES} LINK_LIBRARIES ${PROJECT_LIBS})
 #install resources into DF/stonesense
 install(DIRECTORY resources/
 	DESTINATION "${DFHACK_DATA_DESTINATION}/stonesense"
-- 
1.7.12.4

